nat/service_ics.go

```diff
@@ -113,6 +121,11 @@ func (nat *serviceICS) Del(rule RuleForwarding) error {
 
 // Disable disables internet connection sharing for the public interface.
 func (nat *serviceICS) Disable() (resErr error) {
+	if nat.ipAddrWatchdog != nil {
```

Comment:
These kind of checks and nil assigments are bug sources especially in multiple go routines. Is it really necessary to do it? Just init channel on constructor and close it when neeed

Reply:
We probably should not be able to call 'Disable' if service is not enabled. `ipAddrWatchdog` then never would be nil.

Reply:
The concept must be improved anyway. Maybe watchdog go routine can do all cleanup when signal to stop is received, and exported Disable just closes the channel - that way it would be simpler and without a need for complex synchronization.